Tests/PayID/PayIDUtilsTests.swift

```diff
@@ -2,63 +2,67 @@ import XCTest
 import XpringKit
 
 class PayIDUtilsTest: XCTestCase {
-  func testParsePaymentPointerHostAndPath() {
-    // GIVEN a payment pointer with a host and a path.
-    let rawPaymentPointer = "$example.com/foo"
+  func testParseValidPayID() {
+    // GIVEN a Pay ID with a host and a path.
+    let host = "xpring.money"
+    let path = "georgewashington"
+    let rawPayID = "\(path)$\(host)"
 
-    // WHEN it is parsed to a PaymentPointer object
-    let paymentPointer = PayIDUtils.parse(paymentPointer: rawPaymentPointer)
+    // WHEN it is parsed to components.
+    let payIDComponents = PayIDUtils.parse(payID: rawPayID)
 
     // THEN the host and path are set correctly.
-    XCTAssertEqual(paymentPointer?.host, "example.com")
-    XCTAssertEqual(paymentPointer?.path, "/foo")
+    XCTAssertEqual(payIDComponents?.host, host)
+    XCTAssertEqual(payIDComponents?.path, "/\(path)")
   }
 
-  func testParsePaymentPointerWithWellKnownPath() {
-    // GIVEN a payment pointer with a well known path.
-    let rawPaymentPointer = "$example.com"
+  func testParsePayIDTooManyDollarSigns() {
+    // GIVEN a Pay ID with too many '$'.
+    let host = "xpring$money" // Extra '$'
+    let path = "georgewashington"
+    let rawPayID = "\(path)$\(host)"
 
-    // WHEN it is parsed to a PaymentPointer object
-    let paymentPointer = PayIDUtils.parse(paymentPointer: rawPaymentPointer)
+    // WHEN it is parsed to components.
+    let payIDComponents = PayIDUtils.parse(payID: rawPayID)
 
-    // THEN the host and path are set correctly.
-    XCTAssertEqual(paymentPointer?.host, "example.com")
-    XCTAssertEqual(paymentPointer?.path, PaymentPointer.paymentPointerWellKnownPath)
+    // THEN the Pay ID failed to parse.
+    XCTAssertNil(payIDComponents)
   }
 
-  func testParsePaymentPointerWithWellKnownPathAndTrailingSlash() {
-    // GIVEN a payment pointer with a well known path and a trailing slash.
-    let rawPaymentPointer = "$example.com"
+  func testParsePayIDEmptyHost() {
+    // GIVEN a Pay ID with an empty host.
+    let host = ""
+    let path = "georgewashington"
+    let rawPayID = "\(path)$\(host)"
 
-    // WHEN it is parsed to a PaymentPointer object
-    let paymentPointer = PayIDUtils.parse(paymentPointer: rawPaymentPointer)
+    // WHEN it is parsed to components.
+    let payIDComponents = PayIDUtils.parse(payID: rawPayID)
 
-    // THEN the host and path are set correctly.
-    XCTAssertEqual(paymentPointer?.host, "example.com")
-    XCTAssertEqual(paymentPointer?.path, PaymentPointer.paymentPointerWellKnownPath)
+    // THEN the Pay ID failed to parse.
+    XCTAssertNil(payIDComponents)
   }
 
-  func testParsePaymentPointerIncorrectPrefix() {
-    // GIVEN a payment pointer without a '$' prefix
-    let rawPaymentPointer = "example.com/"
+  func testParsePayIDEmptyPath() {
+    // GIVEN a Pay ID with an empty user.
+    let host = "xpring.money" // Extra '$'
```

Comment:
What do you mean by "Extra '$'"?

Reply:
This is errant from above (the too many dollar signs test)